spdmlib/src/crypto/x509v3.rs

```diff
@@ -139,13 +155,28 @@ fn check_tbs_certificate(
     t_walker += bytes_consumed;
 
     // subjectPublicKeyInfo SubjectPublicKeyInfo,
-    check_public_key_info(&data[t_walker..])?;
+    let bytes_consumed = check_public_key_info(&data[t_walker..])?;
+    t_walker += bytes_consumed;
 
     // issuerUniqueID  [1]  IMPLICIT UniqueIdentifier OPTIONAL,
     // subjectUniqueID [2]  IMPLICIT UniqueIdentifier OPTIONAL,
     // extensions      [3]  EXPLICIT Extensions OPTIONAL
 
-    Ok(length_before_tbs + tbs_length)
+    //key_usage             EXTENSIONS,
+    let (find_key_usage, key_usage_value) = get_key_usage_value(&data[t_walker..])?;
+    let mut check_extensions_success =
+        !(find_key_usage && (LIBSPDM_CRYPTO_X509_KU_DIGITAL_SIGNATURE & key_usage_value == 0x00));
```

Comment:
What does the key usage value 0x00 mean?
if key usage value has just several candidates, I would recommend use Enum for them.

Reply:
Here only need to check key_usage value bit0-digitalSignature. When find the key usage value but get bit0 unset, then return false. It means subject public key in the cert is used for verifying digital signatures in an entity authentication service, a data origin authentication service, and/or an integrity service. Base on RFC5280 page 30-31:

> The digitalSignature bit is asserted when the subject public key is used for verifying digital signatures, other than signatures on certificates (bit 5) and CRLs (bit 6), such as those used in an entity authentication service, a data origin authentication service, and/or an integrity service.

> If the subject public key is only to be used for verifying signatures on certificates and/or CRLs, then the digitalSignature and nonRepudiation bits SHOULD NOT be set.

In libspdm, it doesn't check any other key_usage bits except bit0-digitalSignature. I follow the libspdm logic:
https://github.com/DMTF/libspdm/blob/3bbcb24dbae81a25cb04dd185b0fe948e6ac6b25/library/spdm_crypt_lib/libspdm_crypt_cert.c#L1082-L1093

Bit String Mapping:
https://github.com/Wenxing-hou/libspdm/commit/f9e1dcfdc4d3a2dfcfb05d6b351bc6f28614b504

Reply:
Check only bit0-digitalSignature looks good to me now, one more suggestion:
define a macro for bit0-digitalSignature or add a comment besides it.